### ISSUE TITLE
test killed on travis without test failures 

### ISSUE THREAD

**Issue by manuzhang** [gearpump/gearpump#1611]
```
/home/travis/build.sh: line 45:  2303 Killed                  sbt -jvm-opts project/travis/jvmopts clean coverage +test
```

This almost fails all the travis build.
Has anybody got a clue ?

**Comment by clockfly** [comment]
Maybe timeout.

Below is my guess:
If for a long time, there is no console output, travis will think the job hang and kill it.

**Comment by manuzhang** [comment]
so we should limit the duration of our UT to avoid that ?

**Comment by clockfly** [comment]
@manuzhang 

The problem first appears at PR of #1540, as I commented in the review pannel of #1540.
https://travis-ci.org/gearpump/gearpump/builds/92654958

I think the storm module test introduced this problem.

You can find from the travis, only two storm test is executed

**Comment by manuzhang** [comment]
yes, but later PRs killed on different specs

**Comment by clockfly** [comment]
You can try to add storm test one by one, and see which spec cause the timeout.

**Comment by huafengw** [comment]
I'm also verifying this issue and my suspect is https://github.com/gearpump/gearpump/pull/1555

**Comment by manuzhang** [comment]
travis has a 3G memory limit https://docs.travis-ci.com/user/common-build-problems/#My-build-script-is-killed-without-any-error while we allow for 4G maximum JVM memory.

**Comment by huafengw** [comment]
Fixed in pr #1642

**Comment by stanleyxu2005** [comment]
I highly recommend do a startup check on JVM options and print warnings on unsuitable option values
